[RFR] Get the scoreboard of a grid

benjaminpayoux opened the pull request:
[Trello](https://trello.com/c/pxp0PSNR/1-1-as-peter-given-a-small-board-with-a-single-destroyer-i-want-to-know-every-cell-probability-to-have-the-boat-on-it)

## TODO
- [X] Tests different ships size on 3x3 grid
- [X] Display the scoreboards along tests

## Result
![last](https://user-images.githubusercontent.com/1423027/93860981-21403d80-fcc0-11ea-89dc-b830bc0f13ee.gif)

benjaminpayoux committed:
Get a scoreboard from a grid

JulienMattiussi commented on test/battleship/game/scoreboard_test.go:
```diff
@@ -0,0 +1,35 @@
+
+	actual := scoreboard.GetStringFromScoreBoard(scoreBoard)
+
+	if actual != expected {
```
You should use something like this to be more lisible
https://godoc.org/github.com/stretchr/testify/assert
https://github.com/corbym/gocrest

JulienMattiussi commented on test/battleship/game/scoreboard_test.go:
```diff
@@ -0,0 +1,35 @@
+	// Given one 2 long ship
+	ship := game.Ship{2}
+
+	// When computing its possible positions
```
This is also part of "Given" step

JulienMattiussi commented on test/battleship/game/scoreboard_test.go:
```diff
@@ -0,0 +1,35 @@
+	"testing"
+)
+
+func TestScoreBoardWithOneShip(t *testing.T) {
```
This should identify the tested function like
"TestGetScoreBoard"

benjaminpayoux committed:
Code reviews:

Add tests
Clean tests with gocrest package
Extract ScoreBoard creation and make a struct

mchaffotte commented on src/battleship/grid/grid.go:
```diff
@@ -0,0 +1,51 @@
+	for row := 0; row < gridSize; row++ {
+		for column := 0; column < gridSize; column++ {
+
+			if column+ship.Length <= gridSize {
```
Did you run the command to lint the code ? there is no space around the +

benjaminpayoux replied:
The vscode extension auto format this line and the `make lint` does not complain about it.

mchaffotte commented on src/battleship/scoreboard/scoreboard.go:
```diff
@@ -0,0 +1,36 @@
+	return ScoreBoard{size, squares}
+}
+
+func GetStringFromScoreBoard(scoreBoard *ScoreBoard) string {
```
maybe toString is enough as function name as we know we are using an scoreboard

mchaffotte commented on src/battleship/scoreboard/scoreboard.go:
```diff
@@ -0,0 +1,36 @@
+	Squares [][]int
+}
+
+func MakeScoreBoard(size int) ScoreBoard {
```
you use NewGrid in previous file, why did you change the name in that file MakeScoreBoard ? Be consistent

JulienMattiussi replied:
For me it ok, Grid was the playing board
The score Board is different
Is'nt this ?

benjaminpayoux replied:
Even if the models are different, the methods do the same thing, which is creating a two dimensionnal array. So I agree to stay consistent on methods name.

mchaffotte commented on test/battleship/game/scoreboard_test.go:
```diff
@@ -0,0 +1,73 @@
+
+	// Then it should equals this score board
+	then.AssertThat(t, actual, is.EqualTo(expected).Reason("1 square long ship on 3x3 grid"))
+	DisplayScoreBoard(actual, ship)
```
You can reduce the visibility of that method as it is used only in that file

JulienMattiussi commented on test/battleship/game/scoreboard_test.go:
```diff
@@ -0,0 +1,73 @@
+	_, error := grid.GetScoreBoard(3, ship)
+
+	// Then it should not be computed and return an error
+	if error == nil {
```
Why doesn't this case just return a zero filled scoreBoard ?

benjaminpayoux replied:
Right. A too long ship will not be computed but a score board can still be returned

benjaminpayoux committed:
Rename square to cell

Stay consistent with domain naming

benjaminpayoux committed:
Apply reviews and lint
